src/components/v2/appDetails/k8Resource/StatusFilterButton.component.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useEffect } from "react";
+import IndexStore from "../index.store";
+import { useParams } from 'react-router';
+import { useState } from "react";
+
+export const StatusFilterButtonComponent: React.FC<{}> = ({ }) => {
+    const nodes = IndexStore.getAppDetailsNodes()
+
+    const [tabs, setTabs] = useState([]);
+    const [selectedTab, setSelectedTab] = useState('all')
+    const params = useParams<{ envId: string, appId: string }>()
+
+    const handleFileterClick = (filterName: string) => {
+        IndexStore.updateFilterType(filterName)
+        setSelectedTab(filterName.toLowerCase())
+    }
+
+    useEffect(() => {
+
+        if (nodes.length > 0) {
+            let allNodeCount: number = 0, healthyNodeCount: number = 0, progressingNodeCount: number = 0, failedNodeCount: number = 0
+
+            nodes.map((_node) => {
+                let _nodeHealth = _node.health?.status || ''
+
+                if (_nodeHealth.toLowerCase() === "healthy") {
+                    healthyNodeCount++
+                } else if (_nodeHealth.toLowerCase() === "failed") {
+                    failedNodeCount++
+                } else if (_nodeHealth.toLowerCase() === "progressing") {
+                    progressingNodeCount++
+                }
+                allNodeCount++
+            })
+
+
+            const statusViewTabJSON = [
+                { status: "ALL", count: allNodeCount, isSelected: true },
+                { status: "HEALTHY", count: healthyNodeCount, isSelected: false },
+                { status: "FAILED", count: failedNodeCount, isSelected: false },
+                { status: "PROGRESSING", count: progressingNodeCount, isSelected: false },
+            ]
+
+
+            setTabs(statusViewTabJSON)
+        }
+
+    }, [nodes.length])
```

Comment:
this cannot depend on nodes.length, as status of each node may change over time. Should depend on nodes, may be should subscribe to obeservable